docs/decisions/2022-07-test-fixtures.md

```diff
@@ -0,0 +1,90 @@
+
+## Priorities
+
+Ideally, our test fixtures would be designed such that:
+
+* Test infrastructure is quick to set up.
+* Test infrastructure is documented (e.g. via Terraform files).
+* Tests themselves are quick to write.
+* Both tests and their infrastructure require minimal maintenance.
+* Tests are runnable by multiple [teams of] developers in parallel.
+* Tests cover as much of the application surface as possible.
+* Tests allow developers and contributors to quickly identify root causes of bugs.
+* Tests have minimal reliance on existing state.
+
+Of course, some of these goals are mutually exclusive: supporting parallel test runs often increases the cost of writing (and in some cases maintaining) tests and test infrastructure.
+
+## Constraints
+
+### Some resources MUST be singletons
+While most of Emblem's Google Cloud resources are provisioned programmatically using Terraform, some (such as the Google Cloud projects themselves) are currently intended to be provisioned manually.
+
+> Consequently, we can't rely entirely on automated provisioning systems (at least, not without significant further engineering and/or documentation work) to spin the entire project up and down arbitrarily.
+
+### Emblem is owned by a single team
+While Emblem is designed with a service-oriented architecture, all of Emblem's components are (currently) owned by the same team.
+
+> Consequently, optimizing our test fixtures for multi-team collaboration (such as large numbers of parallel test runs) isn't as important as minimizing test development/maintenance costs.
+
+## Considered Options
+
+In this case, we have a _continuous spectrum_ of options. Thus, the "options" below are more _extremes_ than options:
+
+* **Extreme 1:** All fixtures are centralized, long-lived singletons. Under no circumstances should 2+ instances of the same fixture exist.
+* **Extreme 2:** All fixtures are ephemeral and short-lived, and pluralistic (multiple instances can simultaneously coexist). 
+
+## Decision
+
+We'll use _programmatically provisioned_ fixture resources wherever practical. These resources can be made ephemeral and/or pluralistic if necessary.
+
+Examples include:
+
+- Cloud Run services
+- Cloud Storage buckets
+- Service Accounts
+- IAM policies
+- Artifact Registry repositories
+
+In some situations, _manual provisioning_ of centralized resources will be simpler to implement (if not outright required). Usually, these resources **cannot** be made ephemeral and must be reused between test runs.
+
+Examples include:
+
+- Google Cloud projects
+- Cloud Firestore environments _(these are tied to Google Cloud projects)_
+
+### Rationale
+
+The entire Emblem application is currently owned by a single team. Thus, we decided to weigh the tradeoffs involved accordingly:
+
+- ✅ Cost of development is important
+- ✅ Cost of maintenance is important
+- :x: Parallel test run support is **not** important
```

Comment:
Disagree: This is important. The R&D time has not been prioritized for this, instead we've focused on other forms of technical debt, management capabilities, and general roadmap items. It's relative priority is likely to change over time, either because our circumstances do (e.g., we want to run all forms of testing on every PR because of how we plan maintenance) or because the roadmap of items deemed more important is cleared.

Reply:
Changed from `not important` to `less important`. 👍